instrumentation/failsafe-3.0/library/src/test/java/io/opentelemetry/instrumentation/failsafe/v3_0/FailsafeTelemetryTest.java

```diff
@@ -80,6 +88,82 @@ void captureCircuitBreakerMetrics() {
                                     1, "failsafe.circuit_breaker.state", "closed"))));
   }
 
+  @Test
+  void captureRetryPolicyMetrics() {
+    // given
+    RetryPolicy<Object> userRetryPolicy =
+        dev.failsafe.RetryPolicy.builder()
+            .handleResultIf(Objects::isNull)
+            .withMaxAttempts(3)
+            .build();
+    FailsafeTelemetry failsafeTelemetry = FailsafeTelemetry.create(testing.getOpenTelemetry());
+    RetryPolicy<Object> instrumentedRetryPolicy =
+        failsafeTelemetry.createRetryPolicy(userRetryPolicy, "testing");
+
+    // when
+    for (int i = 0; i <= 4; i++) {
+      int temp = i;
+      AtomicInteger retry = new AtomicInteger(0);
+      Failsafe.with(instrumentedRetryPolicy)
+          .get(
+              () -> {
+                if (retry.get() < temp) {
+                  retry.incrementAndGet();
+                  return null;
+                } else {
+                  return new Object();
+                }
+              });
+    }
+
+    // then
+    testing.waitAndAssertMetrics("io.opentelemetry.failsafe-3.0");
```

Comment:
is there a reason not to use the other style of assertions here? 

```
testing.waitAndAssertMetrics(
        "io.opentelemetry.failsafe-3.0",
        metricAssert ->
            metricAssert
                .hasName("failsafe.retry_policy.execution.count")
... etc
```

Reply:
The usual way we do that @jaydeluca pointed out waits until the assertion succeeds. For example when the method is called and the metric data points aren't available yet (data is exported from a background thread) it will retry the assertions after waiting a bit.
The way you write it `testing.waitAndAssertMetrics("io.opentelemetry.failsafe-3.0");` waits for any metric data to be available. The following code assumes that you have exactly 2 metrics. I find it somewhat hard to reason whether this is guaranteed or not. Probably it is, because no other metrics should be generated, but hard to tell whether there could be something else that could affect this. That is why I believe it is best to write the assertions the same way as they are used elsewhere.